postgres/server/init.sls

```diff
@@ -61,27 +61,43 @@ postgresql-{{ bin }}-altinstall:
 {%- endif %}
 
 postgresql-cluster-prepared:
+  file.directory:
+    - name: {{ postgres.conf_dir }}
+    - user: {{ postgres.user }}
+    - group: {{ postgres.group }}
+    - makedirs: True
+    - recurse:
+      - user
+      - group
+    - dir_mode: 755
   cmd.run:
+ {%- if postgres.prepare_cluster.command is defined %}
+      {# support for depreciated 'prepare_cluster.command' pillar #}
     - name: {{ postgres.prepare_cluster.command }}
+    - unless: {{ postgres.prepare_cluster.test }}
+ {%- else %}
+    - name: {{ postgres.prepare_cluster.pgcommand + ' ' }} {{ postgres.data_dir }}
```

Comment:
I saw you have reconstructed the `pgcommand` defaults... I like that.
But just consider to use short option for all platforms except Debians: `initdb -D`. In such case adding that space in the template would be unnecessary.

Reply:
So I can replace `{{ postgres.prepare_cluster.pgcommand + ' ' }}` with `initdb -D` for all except Debian?